### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by bzgaspar**
Add metadata. Use useRouter instead of window. Add sun icon if light mode

### CODE REVIEW

**Comment by bzgaspar** [comment]
[Preview Link](https://rest-countries-api-git-feature-refinement-itsmenek.vercel.app?_vercel_share=Wcqb0cOzownYDJCPLGavnQi62ErjPeqP)

File: src/app/layout.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ export default function RootLayout({
       <body>
         <ThemeProvider>
           <Header />
-          {children}
         </ThemeProvider>
+        <main>{children}</main>
```
Comment (eremocal) [inline src/app/layout.tsx]: I believe this should be part of `ThemeProvider`?

Comment (bzgaspar) [reply src/app/layout.tsx]: the themeProvider was only used in header part. I thought its okay to not include main. Should I include if thats make it better?

**Comment by kcimartinez** [comment]
Buttons has different font in vercel deployment. 
image

**Review by eremocal — COMMENTED** [review]
When changing to light mode and refreshing the page, the current selected mode goes to initial, which is dark mode.

https://github.com/itsmeNEK/rest-countries-api/assets/61583701/f83c1bd8-2c14-48bb-8bf7-78b2ca033bb3

**Review by eremocal — APPROVED** [review]

**Pull request merged**